test/test_guardrail_globals.py

```diff
@@ -23,92 +15,90 @@ def test_mutable_lint_cmd_empty(self):
         returnval = global_obj.mutable_lint_cmd()
         self.assertEqual(returnval, "")
 
-    def test_generate_pylint_cmd_empty(self):
-        """Function to test generate_pylint_cmd_empty method"""
-        global_obj = GuardrailGlobals()
-        global_obj.linting = None
-        global_obj.mutable_lint_cmd = MagicMock(return_value="xyz")
-        returnval = global_obj.generate_pylint_cmd()
-        self.assertEqual(returnval, "")
-
     def test_generate_pylint_cmd_string(self):
         """Function to test generate_pylint_cmd_string method"""
+        import os
         global_obj = GuardrailGlobals()
         global_obj.linting = "abc"
         global_obj.python = "python"
+        ini_path = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+        ignore_file_path = os.path.join(ini_path, "test_resource", "pylint_ignore.txt")
+        global_obj.lint_ignore = ignore_file_path
+        global_obj.all_folders = "xyz"
         global_obj.mutable_lint_cmd = MagicMock(return_value="xyz")
+        global_obj.generate_files_lint = MagicMock(return_value=" ")
         returnval = global_obj.generate_pylint_cmd()
-        self.assertEqual(returnval, "python -m pylint  xyz")
+        print(returnval)
+        self.assertEqual(returnval[0], "python -m pylint    --output-format=parseable xyz")
 
-    def test_get_exclude_cc_empty(self):
+    @staticmethod
+    def refactor_test_get_exclude_empty(gate, string=None):
         """Function to test get_exclude_cc_empty method"""
         global_obj = GuardrailGlobals()
-        global_obj.cyclo_exclude = None
-        returnval = global_obj.get_exclude_cc()
-        self.assertEqual(returnval, "")
+        retval = None
+        if gate == "cc":
```

Comment:
Are we measuring cyclomatic complexity of our test methods too? If yes, is this within the limit? Could you please help with what limit we have set?

Reply:
Yes code and test is check for CC <10